services/slam/slam.go

```diff
@@ -118,74 +103,3 @@ func GetInternalStateFull(ctx context.Context, slamSvc Service) ([]byte, error)
 	}
 	return HelperConcatenateChunksToFull(callback)
 }
-
-type reconfigurableSlam struct {
-	mu     sync.RWMutex
-	name   resource.Name
-	actual Service
-}
-
-func (svc *reconfigurableSlam) Name() resource.Name {
-	return svc.name
-}
-
-func (svc *reconfigurableSlam) GetPosition(ctx context.Context) (spatialmath.Pose, string, error) {
-	svc.mu.RLock()
-	defer svc.mu.RUnlock()
```

Comment:
[q] Do services need to handle any of this locking & unlocking around their methods or is this now an RDK reconfiguration framework concern?

Reply:
yes they do. they always have needed to

Reply:
Ok, I guess I'm a little confused then b/c it appears as if after this change many services won't have any locking around their grpc calls or Service interface methods:
For example for [motion](https://github.com/viamrobotics/rdk/pull/2225/files/b444fe8109a979ce6eaf6ee77df6fb048752236c#diff-5d0def8554cf047ec606dbb61f57f372934db8240cc93eb0c1b08a5d97012f32L42)  doesn't appear to do any locking around ensuring there are not 2 different move calls happening at the same time:
https://github.com/viamrobotics/rdk/pull/2225/files/b444fe8109a979ce6eaf6ee77df6fb048752236c#diff-01573c2337c9ee8bafdab389eacebcbd7c16692bd71bec1208ead3665f40920fL114

Is that expected?

Reply:
I meant resources need to handle locking where it is important which has always been the case. a RLock was only mutexing the resource access for when swapping happened. Motion would already have this issue as described. This did not change that. Sounds like a bug.

Reply:
That particular one is a known issue: https://viam.atlassian.net/browse/RSDK-1233